### ISSUE TITLE
Dot files are copied from assets to build

### ISSUE THREAD

**Issue by evanmiller** [scmorrison/uzu#32]
I noticed that e.g. Vim swap files in `assets` are being copied over to my build directory.

```
        build/ab-testing/.abtools.css.swp
        build/ab-testing/.sample-size.css.swp
```

**Comment by scmorrison** [comment]
Yeah, uzu copies `public/` and `themes/**/assets/` wholesale. The reason for this is that the user might have a hiddent file that they want to keep within their uzu project.

I use vim too and I use something like this im my `.vimrc` that moved the backup dir. The .swp files will be saved in there instead.

```
set backupdir=~/.vim/backup//
set directory=~/.vim/swp//
```

Hope that helps. I have added a [note](https://github.com/scmorrison/uzu/tree/dev#public-and-assets-directories) to the README about this too.

**Comment by evanmiller** [comment]
Ok. In the future a config option that behaves similarly to `.gitignore` would be nice.

**Comment by scmorrison** [comment]
It would be useful. Let's use this issue to capture ideas around the config option for now. Maybe something like this:

```yaml
exclude_files:
    - '*.swp'              # any files with .swp extension
    - 'path/to/dir'        # specific directory
    - 'path/**/tmp'        # any folders called tmp inside of any dir under path
    - 'path/to/file'       # specific file
    - '*string-to-match*'  # any dir / file with string-to-match in name
```

I would also like to make the copy behavior more rsync-like. 

Right now using `cp` is fast. Whatever replaces it needs to be relatively speedy (and work on Windows) so builds don't get hit with huge file copy cost for each build.

**Comment by evanmiller** [comment]
I like the `exclude_files` ideas. At least for my projects, I doubt that custom file copy logic will begin to compare to the template rendering times described in #10, so I'm in favor of using an exclusion list.

**Comment by scmorrison** [comment]
@evanmiller I have added an `exclude` option that allows you to specify file names and folder names to exclude from copying.

```
# List of directories and files to exclude from build/
exclude:
  - node_modules
  - packages.json
  - yarn.lock
```

It will exclude any file or folder that matches full / partially any of the strings in that list.

**Comment by evanmiller** [comment]
@scmorrison Thanks. Does globbing work with this feature? My original goal was to exclude files matching `.*.swp`

**Comment by scmorrison** [comment]
No, not yet. But something like `.swp` should work to catch anything with .swp extension.
